src/modules/ekf2/ekf2_main.cpp

```diff
@@ -951,7 +952,7 @@ void Ekf2::run()
 						}
 					}
 
-					_ekf.setRangeData(range_finder.timestamp, range_finder.current_distance);
+					if (range_finder_updated) { _ekf.setRangeData(range_finder.timestamp, range_finder.current_distance); }
```

Comment:
We don't want to change this line. I twill cause loss of height estimation if range finder is a primary height source and the range sensor goes out of range when on ground due to the vehicle being hand carried, etc.

Reply:
OK, but can you then propose an alternative solution?

The current structure just does not make sense in my opinion. We set `range_finder_updated = false;` in line 948, but we never use that information anywhere in the code! Effectively, this means that we are not obeying the range_finder.min_distance and range_finder.max_distance values delivered by the drivers. I did see that if rangeSensor=primary height source, then we use those limits as defined [here](https://github.com/PX4/ecl/blob/41953ab5823fe0e6d062e3641c37f5c05a3e5cc2/EKF/ekf_helper.cpp#L1086), but if we only use the range sensor for terrain estimation, then we'll always use range data even if its beyond the device limitations, yielding the issues seen in https://github.com/PX4/Firmware/issues/9843 . Do you see what i mean?

Reply:
The prior structure was logical. The code used a value equal to EKF2_MIN_RNG if the range finder returned an invalid reading when the vehicle was on the ground. Invalid readings during on-ground operation due to sensor obstruction from grass, operator handling, sonar reflections, etc were/are very common.

It is important that this PR does not change that behaviour.

Reply:
I see what you mean. The result of separate changes over the past few months in the ekf2 module and ecl library is that although min and max range is passed through to the ekf here https://github.com/PX4/Firmware/blob/master/src/modules/ekf2/ekf2_main.cpp#L957, the ekf does not use these limits internally to reject data.

I will submit a bug fix for the ecl library.

Reply:
See https://github.com/PX4/ecl/pull/476

Reply:
The upstream bug fix has now been merged. I will now submit a change request to get all the  checking, including the decision to use the on ground value into the one function inside the ecl library so that we don't end up again in a situation where having checks in multiple places results in functionality being lost.